pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -304,6 +312,64 @@ synchronized void runIncrementalPipeline()
     }
   }
 
+  /**
+   * This method periodically checks on Thrift server output directory and commands to create
+   * resource tables. The very first run happens over a minute later once controller spins up. And
+   * it's programmed to check on Thrift server every hour in case it went down for unfortunate
+   * reasons.
+   *
+   * @throws IOException
+   * @throws SQLException
+   */
+  @Scheduled(initialDelay = 60000, fixedDelay = 3600000)
+  private void createResourceTablesOnStart() throws IOException, SQLException {
+    if (!dataProperties.isCreateHiveResourceTables()) {
```

Comment:
It's also good to add a condition to have only one instance of the job running at any point in time. Can you also please check if there will be any race conditions when the batch/incremental pipelines are running.

Reply:
Yeah that's actually one reason that I think we should avoid the scheduled option for now. So one thread might be updating tables after a successful pipeline run, while another thread is here doing the scheduled check.

We can keep this function the way it is but instead of annotating it with `@Scheduled` call it at the end of `initDwhStatus`, i.e., only once at `@PostConstruct`.